.github/workflows/main.yml

```diff
@@ -41,6 +41,24 @@ jobs:
     - name: Run Ruby tests
       run: bundle exec rake compile_no_debug
 
+  build-debug-mode:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: head
+        bundler-cache: true
+    - name: Run Ruby tests
+      run: bundle exec rake
+      env:
+        YARP_DEBUG: "1"
```

Comment:
I think the general approach here is totally fine, but I'm unclear on `YP_DEBUG` versus `YARP_DEBUG` now. Could we either name this something different or make `YP_DEBUG` have other options?